### PR TITLE
WDFT MARC 2022 - RUBEN POELEN

### PR DESCRIPTION
**Pull request by coderKron**
did everything that was needed except for bonusses, might do them if im in the mood this evening.

### CODE REVIEW

File: src/functions-and-arrays.js [hunk 1]
```diff
@@ -1,38 +1,104 @@
...
+    } else if (word.length === arr[i].length) {
+      continue;
+    }
+  }
+  return word;
+}
 
+const longestWord = findLongestWord([]);
 
+console.log(longestWord);
```
Comment (TA-Remote) [inline src/functions-and-arrays.js:38]: After you finish writing code, try to always clean up and remove any lines that are unused ;)

File: src/functions-and-arrays.js [hunk 2]
```diff
@@ -1,38 +1,104 @@
...
+function findLongestWord(arr) {
+  let word = '';
+  if (arr === undefined) {
+    return null;
+  } else if (arr.length === 0) {
+    return null;
+  }
+  for (let i = 0; i < arr.length; i++) {
+    if (arr[i].length === 0) {
+      return null;
```
Comment (TA-Remote) [inline src/functions-and-arrays.js:26]: what are you checking here?

File: src/functions-and-arrays.js [hunk 3]
```diff
@@ -52,16 +118,45 @@ const wordsUnique = [
...
+function doesWordExist(arrOfWords, word) {
+  if (arrOfWords.length === 0) {
+    return null;
+  }
+  for (let i = 0; i < arrOfWords.length; i++) {
+    if (arrOfWords[i].indexOf(word) !== -1) {
+      return true;
+    } else {
+      continue;
+    }
```
Comment (TA-Remote) [inline src/functions-and-arrays.js:153]: with if statements you do not need to use `continue`, also in previous iterations.
You are inside the `for loop`, so you are already iterating through each element of the array , and only exit the loop if you find exisitng word in this case

File: src/functions-and-arrays.js [hunk 4]
```diff
@@ -52,16 +118,45 @@ const wordsUnique = [
...
+  }
+  return noDuplicates;
+}
 
+const noDuplicates = uniquifyArray(wordsUnique);
+console.log(noDuplicates);
 
 // Iteration #6: Find elements
```
Comment (TA-Remote) [inline src/functions-and-arrays.js:141]: Iteration 6 task says, ` Don't use indexOf for this one.`. Can you think of an alternative way to solve this ? ;)

**Review by TA-Remote — COMMENTED** [review]
Great work Ruben, keep it up!

File: src/functions-and-arrays.js (hunk 4 above)
Comment (coderKron) [reply src/functions-and-arrays.js:141]: function doesWordExist(arrOfWords, word) {
  if (arrOfWords.length === 0) {
    return null;
  }
  for (let i = 0; i < arrOfWords.length; i++) {
    if (arrOfWords[i].includes(word) === true) {
      return true;
    }
  }
  return false;
}

Comment (coderKron) [reply src/functions-and-arrays.js:141]: I read it wrong oops!

**Pull request closed without merging**